File: 系列1-MyBatis源码解析/第12节课/mybatis/src/main/java/com/enjoylearning/mybatis/config/Configuration.java
/**
 *    Copyright ${license.git.copyrightYears} the original author or authors.
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */
package com.enjoylearning.mybatis.config;

import java.util.HashMap;
import java.util.Map;

import com.enjoylearning.mybatis.binding.MapperProxyFactory;
import com.enjoylearning.mybatis.session.SqlSession;

/**
 * @author Clinton Begin
 */
public class Configuration {
	//记录mapper xml文件存放的位置
	public static final String MAPPER_CONFIG_LOCATION = "config";
	//记录数据库连接信息文件存放位置
	public static final String DB_CONFIG_FILE = "db.properties";
    
	private String dbUrl;

	private String dbUserName;

	private String dbPassword;

	private String dbDriver;

    //mapper xml解析完以后select节点的信息存放在mappedStatements
	protected final Map<String, MappedStatement> mappedStatements = new HashMap<String, MappedStatement>();
	
	//为mapper接口生成动态代理的方法
	public <T> T getMapper(Class<T> type, SqlSession sqlSession) {
		return MapperProxyFactory.getMapperProxy(sqlSession, type);
	}
	

	public Map<String, MappedStatement> getMappedStatements() {
		return mappedStatements;
	}

	public MappedStatement getMappedStatement(String sourceId) {
		return mappedStatements.get(sourceId);
	}



	public String getDbUrl() {
		return dbUrl;
	}

	public void setDbUrl(String dbUrl) {
		this.dbUrl = dbUrl;
	}

	public String getDbUserName() {
		return dbUserName;
	}

	public void setDbUserName(String dbUserName) {
		this.dbUserName = dbUserName;
	}

	public String getDbPassword() {
		return dbPassword;
	}

	public void setDbPassword(String dbPassword) {
		this.dbPassword = dbPassword;
	}

	public String getDbDriver() {
		return dbDriver;
	}

	public void setDbDriver(String dbDriver) {
		this.dbDriver = dbDriver;
	}

	@Override
	public String toString() {
		return "Configuration [dbUrl=" + dbUrl + ", dbUserName=" + dbUserName
				+ ", dbPassword=" + dbPassword + ", dbDriver=" + dbDriver
				+ ", mappedStatements=" + mappedStatements + "]";
	}

}

File: 系列3-解决方案专题/09-ELK集群搭建实践-赵云/ELK集群搭建快速实践/kibana-5.2.0-linux-x86_64/src/server/logging/configuration.js
/*
 * Copyright 2021-2022 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
exports['default'] = loggingConfiguration;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

var _log_reporter = require('./log_reporter');

var _log_reporter2 = _interopRequireDefault(_log_reporter);

function loggingConfiguration(config) {
  var events = config.get('logging.events');

  if (config.get('logging.silent')) {
    _lodash2['default'].defaults(events, {});
  } else if (config.get('logging.quiet')) {
    _lodash2['default'].defaults(events, {
      log: ['listening', 'error', 'fatal'],
      request: ['error'],
      error: '*'
    });
  } else if (config.get('logging.verbose')) {
    _lodash2['default'].defaults(events, {
      log: '*',
      ops: '*',
      request: '*',
      response: '*',
      error: '*'
    });
  } else {
    _lodash2['default'].defaults(events, {
      log: ['info', 'warning', 'error', 'fatal'],
      response: config.get('logging.json') ? '*' : '!',
      request: ['info', 'warning', 'error', 'fatal'],
      error: '*'
    });
  }

  var options = {
    opsInterval: config.get('ops.interval'),
    requestHeaders: true,
    requestPayload: true,
    reporters: [{
      reporter: _log_reporter2['default'],
      config: {
        json: config.get('logging.json'),
        dest: config.get('logging.dest'),
        // I'm adding the default here because if you add another filter
        // using the commandline it will remove authorization. I want users
        // to have to explicitly set --logging.filter.authorization=none or
        // --logging.filter.cookie=none to have it show up in the logs.
        filter: _lodash2['default'].defaults(config.get('logging.filter'), {
          authorization: 'remove',
          cookie: 'remove'
        })
      },
      events: _lodash2['default'].transform(events, function (filtered, val, key) {
        // provide a string compatible way to remove events
        if (val !== '!') filtered[key] = val;
      }, {})
    }]
  };
  return options;
}

module.exports = exports['default'];
